### PR TITLE
[CALCITE-4207] Validation fails for positional aggregate with current_date in 'case' expression

### PR DESCRIPTION
**Pull request by ihuzenko**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java [hunk 1]
```diff
@@ -11685,6 +11685,19 @@ private void checkCustomColumnResolving(String table) {
...
 
+  @Test void testPositionalAggregateWithExpandedCurrentDateFunction() {
+    SqlConformance defaultPlusOrdinalGroupBy =
+        new SqlDelegatingConformance(SqlConformanceEnum.DEFAULT) {
+          @Override public boolean isGroupByOrdinal() {
+            return true;
+          }
+        };
+    sql("SELECT HIREDATE >= CURRENT_DATE, COUNT(*) "
+        + "FROM EMP GROUP BY 1")
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:11696]: I'm confused by why the `current_date` caused the problem, does other non-param function triggers the bug too ?

Comment (ihuzenko) [reply core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:11696]: Please take a look at the stack trace from [CALCITE-4207](https://issues.apache.org/jira/browse/CALCITE-4207) description and [this comment](https://issues.apache.org/jira/browse/CALCITE-4207?focusedCommentId=17189063&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17189063).

Comment (danny0405) [reply core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:11696]: > In one place CURRENT_DATE was represented as SqlBasicCall while in other it was instance of SqlIdentifier

Still does not got your point, in you fix, you just use the expanded group expr for validation, but the `CURRENT_DATE` is not a group expr ?

Comment (ihuzenko) [reply core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:11696]: Positional aggregate points to the first expression in select. Validation tries to match 2 expression trees : 
1. The first item from select which is already expanded at the moment and contains current_date as an instance of SqlBasicCall 
2. An item from the group by where prior to my fix current_date was presented as an instance of the SqlIdentifier.

Due to this instance mismatch validation goes further and tells that `Expression 'EMP.HIREDATE' is not being grouped`, although grouping was performed by the whole expression `HIREDATE >= CURRENT_DATE`.

**Comment by ihuzenko** [comment]
Hello @zabetak , 

Could you please take a look at the PR? 

Thank you in advance, 
Igor

File: core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java (hunk 1 above)
Comment (ihuzenko) [reply core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:11696]: Hello @danny0405 , could you proceed with reviewing the PR?

Comment (danny0405) [reply core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:11696]: I tried the fix locally and think the fix is reasonable.

**Review by danny0405 — APPROVED** [review]
+1

File: core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java (hunk 1 above)
Comment (ihuzenko) [reply core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java:11696]: Thanks for review:)

**Comment by amaliujia** [comment]
+1

**Pull request merged**